src/lib.rs

```diff
@@ -2,100 +2,76 @@
  # stm32f3xx-hal
 
  `stm32f3xx-hal` contains a multi device hardware abstraction on top of the
- peripheral access API for the STMicro [STM32F3][stm] series microcontrollers. The
- selection of the MCU is done by [feature][f] gates
+ peripheral access API for the STMicro [STM32F3][stm] series microcontrollers.
 
- [f]: #selecting-the-right-chip
  [stm]: https://www.st.com/en/microcontrollers-microprocessors/stm32f3-series.html
 
- # Selecting the right chip
-
-   This crate requires you to specify your target chip as a feature.
-
-   Please select one of the following
-
-   (Note: `x` denotes any character in [a-z])
-   *   stm32f301
-   *   stm32f318
-   *   stm32f302xb
-   *   stm32f302xc
-   *   stm32f302xd
-   *   stm32f302xe
-   *   stm32f302x6
-   *   stm32f302x8
-   *   stm32f303xb
-   *   stm32f303xc
-   *   stm32f303xd
-   *   stm32f303xe
-   *   stm32f303x6
-   *   stm32f303x8
-   *   stm32f373
-   *   stm32f378
-   *   stm32f334
-   *   stm32f328
-   *   stm32f358
-   *   stm32f398
-
-   Example: The STM32F3Discovery board has a STM32F303VCT6 chip.
-   So you want to expand your call to `cargo` with `--features stm32f303xc`.
-
-   For more information, see the [README](https://github.com/stm32-rs/stm32f3xx-hal/blob/master/README.md#selecting-the-right-chip)
+ ## cargo features
+
+ ### Target chip selection
+
+ This crate requires you to specify your target chip as a feature.
+
+ Please select one of the following (`x` denotes any character in [a-z]):
+
+ * stm32f301x6
+ * stm32f301x8
+ * stm32f318x8
+ * stm32f302x6
+ * stm32f302x8
+ * stm32f302xb
+ * stm32f302xc
+ * stm32f302xd
+ * stm32f302xe
+ * stm32f303x6
+ * stm32f303x8
+ * stm32f303xb
+ * stm32f303xc
+ * stm32f303xd
+ * stm32f303xe
+ * stm32f328x8
+ * stm32f358xc
+ * stm32f398xe
+ * stm32f373x8
+ * stm32f373xb
+ * stm32f373xc
+ * stm32f378xc
+ * stm32f334x4
+ * stm32f334x6
+ * stm32f334x8
+
+ Example: The STM32F3Discovery board has a STM32F303VCT6 chip.
+ So you need to specify `stm32f303xc` in your `cargo.toml` (note that VC → xc).
+
+ For more information, see the [README][].
+
+ [README]: https://github.com/stm32-rs/stm32f3xx-hal/blob/master/README.md#selecting-the-right-chip
+
+ ### ld
```

Comment:
Pretty minor nit (can be ignored) I prefer an empty line after a section header.